libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/RoutesObserver.kt

```diff
@@ -28,10 +28,13 @@ fun interface RoutesObserver {
  * The route at index 0, if exist, will be treated as the primary route for 'Active Guidance'.
  *
  * @param navigationRoutes list of currently maintained routes
+ * @param ignoredRoutes list of alternative routes that were ignored
+ *  because they are invalid for navigation. See [IgnoredRoute] for details.
  * @param reason why the routes have been updated (re-route, refresh route, and etc.)
  */
 class RoutesUpdatedResult internal constructor(
```

Comment:
Is this intentional that `RoutesUpdatedResult` does not override equals/hashCode/toString?
@LukasPaczos @VysotskiVadim @RingerJK

Reply:
I don't see reasons to not not override equals/hashCode/toString. I think we have just forgotten 🙂

Reply:
I can think of a reason. `RoutesUpdatedResult` is delivered very frequently, on the other side it contains a list of `NavigationRoute`. And comparing routes is expensive. Maybe we wanted to ensure that if someone does `if (newRoutesUpdatedResult != oldRoutesUpdatedResult)` in `RouteProgressObserver` we don't do any extensive work.
I'm almost sure that's not the case but who knows...

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/6761